plugins/inputs/qbittorrent/qbittorrent_test.go

```diff
@@ -0,0 +1,154 @@
+package qbittorrent
+
+import (
+	"fmt"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestGetMainData_URL(t *testing.T) {
+	fakeServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/api/v2/sync/maindata" && r.Method == "GET" {
+			w.WriteHeader(http.StatusOK)
+			_, err := fmt.Fprintln(w, mainDataJSON)
+			require.NoError(t, err)
+		}
+	}))
+	defer fakeServer.Close()
+
+	plugin := &QBittorrent{
+		URL: fakeServer.URL,
+		cookie: []*http.Cookie{{
+			Name:  "cookieName",
+			Value: "cookieValue",
+		}},
+	}
+
+	var acc testutil.Accumulator
+	require.NoError(t, acc.GatherError(plugin.Gather))
+
+	require.Len(t, acc.Metrics, 2)
+
+	categoriesCount, _ := acc.Int64Field("qbittorrent", "category_count")
+	require.Equal(t, int64(1), categoriesCount)
+	tagsCount, _ := acc.Int64Field("qbittorrent", "tag_count")
+	require.Equal(t, int64(1), tagsCount)
+	freeSpaceOnDisk, _ := acc.Int64Field("qbittorrent", "free_space_on_disk")
+	require.Equal(t, int64(1921000090000), freeSpaceOnDisk)
+	dlInfoData, _ := acc.Int64Field("qbittorrent", "dl_info_data")
+	require.Equal(t, int64(1594136974832), dlInfoData)
+	upInfoData, _ := acc.Int64Field("qbittorrent", "up_info_data")
+	require.Equal(t, int64(300010001000), upInfoData)
+	totalBuffersSize, _ := acc.Int64Field("qbittorrent", "total_buffers_size")
+	require.Equal(t, int64(524288), totalBuffersSize)
+
+	addedOnValue, _ := acc.Int64Field("torrent", "added_on")
+	require.Equal(t, int64(1701232564), addedOnValue)
+	etaValue, _ := acc.Int64Field("torrent", "eta")
+	require.Equal(t, int64(8640000), etaValue)
+	downloadedSessionValue, _ := acc.Int64Field("torrent", "downloaded_session")
+	require.Equal(t, int64(483696196), downloadedSessionValue)
+
+	require.Equal(t, "003fyd3cgjsdf6f8debc4273s4g80bhgb25hs487", acc.TagValue("torrent", "hash"))
+	require.Equal(t, "https://xxxxx/announce.php?passkey=xxxxx", acc.TagValue("torrent", "tracker"))
+	require.Equal(t, "/downloads/c1/torrent1", acc.TagValue("torrent", "content_path"))
```

Comment:
This whole section can be made easier to read by just doing this:

```suggestion
	testutil.RequireMetricsEqual(t, expected, acc.GetTelegrafMetrics())
```